Add new colab notebook for AdalFlow data class implementation

ajithvcoder opened the pull request:
Added a new notebook for dataclass tutorial based on following issues

- https://github.com/SylphAI-Inc/AdalFlow/issues/194
- https://github.com/SylphAI-Inc/AdalFlow/issues/123

ajithvcoder commented:
@liyin2015 let me know if the notebook is fine

Sylph-AI commented on notebooks/adalflow_dataclasses.ipynb:
```diff
@@ -0,0 +1,794 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #5.        box_office: Optional[float] = field(
</pre><p>No need to use Optional. As any field in dataclass that comes with <code>default</code> or <code>default_factory</code> is an optional field. Sometimes using <code>Optional</code> will confuse the LLM or unable to support complicated data structure   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SylphAI-Inc/AdalFlow/pull/226/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

ajithvcoder replied:
i have addressed the issue

Sylph-AI commented on notebooks/adalflow_dataclasses.ipynb:
```diff
@@ -0,0 +1,794 @@
+{
```
<p>Try to use &lt;USER&gt;&lt;/USER&gt; to replace User: This fits more on how LLMs are using special tokens</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SylphAI-Inc/AdalFlow/pull/226/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

ajithvcoder replied:
i have addressed the issue

liyin2015 commented:
@ajithvcoder This is great work to show more complicated dataclass structure. I have left comments and basically need to add "No optional" in the dataclass. And please adjust to this new info and rerun the test again. And then it should be good to go!

liyin2015 reviewed (changes requested):
check out comments

ajithvcoder commented:
@liyin2015  @Sylph-AI  i will do the required changes but it wont pass the ci/cd because there is no quotes [here](https://github.com/SylphAI-Inc/AdalFlow/blob/main/poetry.lock#L6131) . i have addressed those issues in this pr https://github.com/SylphAI-Inc/AdalFlow/pull/227

ajithvcoder commented:
@liyin2015  i have made the changes kindly review, its a single file which is not in Adalflow/main branch so i didnt rebase it. Also it has successfully passed the checks

ajithvcoder commented:
found a method for rebasing and i have rebased it also

liyin2015 reviewed (approved):
looking good.
